src/fem_solver/src/main_matrix_solver.cpp

```diff
@@ -2,7 +2,36 @@
 
 #include <iostream>
 
-SolverHeatmap MainMatrixSolver::ComputeHeatmap(const MatrixEquation& db) {
-	std::cout << "Solving main matrix, generating heatmap..." << std::endl;
-	return SolverHeatmap();
+void CustomPrintMatrix(const Matrix& matrix, const std::string& matrix_name) {
+	std::cout << "Matrix: " << matrix_name << std::endl;
+	for (unsigned i = 0; i < matrix.size1(); ++i) {
+		for (unsigned j = 0; j < matrix.size2(); ++j)
+			std::cout << matrix(i, j) << "\t";
+		std::cout << std::endl;
+	}
+	std::cout << "===END OF MATRIX==========" << std::endl;
+}
+
+const SolverHeatmap& MatrixEquation::Solve() {
+	assert(!already_solved_ && "Can be solved only one time");
+	already_solved_ = true;
+	Matrix& A = coeficients_;
+	Matrix& y = result_;
```

Comment:
please replace A and y with original variables, there is no reason for such move